lib/utils/files.js

```diff
@@ -38,5 +37,13 @@ files.make_artifacts_folder = function (folder_name) {
   return folder_name;
 };  // Ends files.make_artifacts_folder()
 
+files.below_21 = function (page_id) {
+  return `${ page_id }`.substring(0, 20);
+};
+
+files.below_71 = function (scenario_name) {
+  return `${ scenario_name }`.substring(0, 70);
+};
```

Comment:
It took me a bit, but I do understand why these exist (since the inputs might be null, you need to wrap it in a `${ ... }`, which is hard to do nested).  I'd still lean towards just making one function instead of two, and not hard coding the name of the function to exact numbers.

```suggestion
files.below_length = function (length, input) {
  return `${ input }`.substring(0, Math.max(length - 1, 0));
};
```

Reply:
This is a good question to bring up and I agree we might need to change the names. I do think it should stay 2 separate functions. Not sure quite how to articulate this atm, but I'll do my best!

There are two lengths we want to limit to. One is the `id` name max length, which is shorter, which we decided was 21 chars but we might change. The other is the scenario name max length. I don't want to decide at each location where these functions are being called how many characters to allow. If we have to update it, we're then stuck updating in multiple places. Also, it lets someone in the future come by and just change it without much thought (which makes me think these need function descriptions too). I think it might also be possible to name these well enough to clarify what they're being used for, though I can't think of the names right now. I decided against `truncate_to_id_max_length` etc. is because I think that when the scenario description isn't in the mix of the path name, maybe we could allow the question id to be longer. Or other such things.

Does that explain a bit? At least about why they're separate and very specific functions.

Reply:
Thanks, these explanations help a lot.

There are some conflicting arguments:

> I don't want to decide at each location where these functions are being called how many characters to allow. If we have to update it, we're then stuck updating in multiple places. Also, it lets someone in the future come by and just change it without much thought (which makes me think these need function descriptions too).

This argues towards the existing code, just with different names (TBH `truncate_to_id_max_length` is good here).

> I decided against truncate_to_id_max_length etc. is because I think that when the scenario description isn't in the mix of the path name, maybe we could allow the question id to be longer.

This argues towards deciding the length at each call site; if how long we're truncating things to depends on how we're using it, that should be decided at the call site; anything else is gonna be a lot of extra functions and abstractions for limited benefit.

You've done a lot of great work at making choices and shortening things where you can, but IMO we're going to have to tweak the exact numbers, and I'd lean towards making it easier to do that.

Reply:
Those are fair points. In that case, I think I should get rid of these functions because it's just recreating `.substring()`. A couple less functions to maintain! Yay! Also, I believe `.substring()` doesn't need us to limit the length - it automatically uses whichever is shorter.

Reply:
Adjusted. If you have a mo, I'd appreciate your eye on the changes.